### DISCUSSION TITLE
How to use with Meta StyleX

### DISCUSSION THREAD

**Discussion by Tech-Dex** [chakra-ui/ark#1932] [Q&A]
Hello,

I've started a discussion on StyleX github as well around the question on "How to use with a component library (https://github.com/facebook/stylex/discussions/224)" .

They gave a great response, but it is a bit hacky as they state. I was wondering if ARK-ui plans to allow a easy use of StyleX without the hacky way? I am interested on how to achieve the conditional styling.

**Comment by cschroeter** [comment]
When it comes down to styling Ark UI components you can use different appraoches. For example you can simply just add class names to components or write CSS styles that targets  Ark components, as documented here https://ark-ui.com/docs/styling/overview

Looking into the mentioned discussion `data-` attributes are a common way in HTML / CSS to add custom attributes to nodes. So it is up the the Styling Framework to find ways to allow to target this. Since `stylex` is a fairly new framework I recommed more mature options like PandaCSS, Uno CSS or Tailwind to start with.
